pkg/netstore/netstore.go

```diff
@@ -57,31 +66,70 @@ func (s *store) Get(ctx context.Context, mode storage.ModeGet, addr swarm.Addres
 				go s.recoveryCallback(addr, targets)
 				return nil, ErrRecoveryAttempt
 			}
-			stamp, err := ch.Stamp().MarshalBinary()
-			if err != nil {
-				return nil, err
-			}
+			s.wg.Add(1)
+			go func() {
+				defer s.wg.Done()
```

Comment:
wouldn't it make more sense to factor this out to a separate method on `netstore`?